From af0ab01f266a00aa3d8586ad2655a99c79f5f2ab Mon Sep 17 00:00:00 2001
Message-ID: <af0ab01f266a00aa3d8586ad2655a99c79f5f2ab.1708471786.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1708471786.git.sharlatanus@gmail.com>
References: <cover.1708471786.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 27 Jan 2024 22:21:12 +0000
Subject: [PATCH 32/46] gnu: python-photutils: Speed up tests

* gnu/packages/astronomy.scm (python-photutils): Enable parallel
tests to speed them up.
[arguments] <#:test-flags>: Add it with "-n" option.
[native-inputs]: Add python-pytest-xdist.

Change-Id: I3eff6c4ac4f6c3d8367179e1e8cdf55bbf1c03cc
---
 gnu/packages/astronomy.scm | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index a3dca00664..111599f442 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2217,6 +2217,8 @@ (define-public python-photutils
     (build-system pyproject-build-system)
     (arguments
      (list
+      #:test-flags
+      #~(list "-n" "auto")
       #:phases
       #~(modify-phases %standard-phases
           ;; This file is opened in both install and check phases.
@@ -2247,6 +2249,7 @@ (define-public python-photutils
      (list python-cython
            python-extension-helpers
            python-pytest-astropy
+           python-pytest-xdist
            python-semantic-version
            python-setuptools-scm))
     (home-page "https://github.com/astropy/photutils")
-- 
2.41.0

